src/io/trajectory.cpp

```diff
@@ -0,0 +1,98 @@
+#include "aikido/io/trajectory.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <boost/program_options.hpp>
+#include "aikido/common/Spline.hpp"
+#include "aikido/io/yaml.hpp"
+#include "aikido/io/detail/yaml_extension.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+using aikido::statespace::ConstStateSpacePtr;
+using aikido::statespace::StateSpacePtr;
+using aikido::statespace::dart::MetaSkeletonStateSpacePtr;
+using aikido::trajectory::Spline;
+using aikido::trajectory::UniqueSplinePtr;
+
+namespace aikido {
+namespace io {
+
+void saveTrajectory(const aikido::trajectory::Spline& trajectory,
+    const MetaSkeletonStateSpacePtr& skelSpace, const std::string& savePath)
```

Comment:
As above, I think you should be able to just pull the statespace from the trajectory. I would just try to cast it to a `MetaSkeletonStateSpace` to get the properties you need.

Reply:
If you do the above, I suggest using `MetaSkeletonStateSpace` specific functions to make the log/exp conversions.

Reply:
@aditya-vk  Is there any particular reason for that? Because I think using MetaSkeletonStateSpace specific function would first require type-casting the state that we pass to these functions.